client/src/sagas/patient.js

```diff
@@ -163,8 +157,79 @@ function* prescriptionChangeStatus(action) {
   }
 }
 
+function* addParent(action) {
+  try {
+    const { parentId, parentType } = action.payload;
+    const parsedPatient = yield select((state) => state.patient.patient.parsed);
+    const originalPatient = yield select((state) => state.patient.patient.original);
+
+    const patientToUpdate = JSON.parse(JSON.stringify(originalPatient));
+
+    patientToUpdate.extension.push({
+      url: 'http://fhir.cqgc.ferlab.bio/StructureDefinition/family-relation',
+      extension: [
+        {
+          url: 'subject',
+          valueReference: { reference: `Patient/${parentId}` },
+        },
+        {
+          url: 'relation',
+          valueCodeableConcept: {
+            coding: [{
+              system: 'http://terminology.hl7.org/ValueSet/v3-FamilyMember',
+              code: parentType,
+            }],
+          },
+        },
+      ],
+    });
+
+    yield updatePatient(patientToUpdate);
+
+    yield Api.addPatientToGroup(parsedPatient.familyId, parentId, parentType);
```

Comment:
Two things:
* This logic should be moved out from the saga function.
* I can see that we link the parent to the patient, but what about linking the patient to the parent?